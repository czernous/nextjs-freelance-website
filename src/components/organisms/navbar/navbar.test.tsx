import React from 'react';
import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from './index';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('Navbar', () => {
  let navbar;

  beforeEach(async () => {
    navbar = render(
      <Navbar
        test-id="navbar"
        navItems={[
          { uuid: '1', text: 'Home', url: '/' },
          { uuid: '2', text: 'About', url: '/about' },
        ]}
        drawerWidth={200}
      />,
    );
  });

  it('passes axe audit', async () => {
    const { container } = navbar;
    await act(async () => {
      expect(await axe(container)).toHaveNoViolations();
    });
  });

  it('is present in layout', () => {
    const navbar = screen.findByTestId('navbar');
    waitFor(() => expect(navbar).toBeInTheDocument());
  });

  it('is rendered unchanged', () => {
    const { container } = navbar;
    expect(container).toMatchSnapshot();
  });

  it('opens mobile menu(drawer)', async () => {
    await act(async () => {
      const { container } = navbar;
      const button = await container.querySelector('[test-id="drawerButton"]');
      waitFor(() => expect(button).toBeInTheDocument());
      fireEvent(
        button,
        new MouseEvent('click', { bubbles: true, cancelable: true }),
      );
    });
  });
});
