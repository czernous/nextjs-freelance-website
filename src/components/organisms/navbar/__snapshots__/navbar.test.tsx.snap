// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar is rendered unchanged 1`] = `
<div>
  <nav
    aria-label="Main"
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed appbar mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    role="navigation"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular custom-container css-hyum1k-MuiToolbar-root"
      style="width: 100%; height: 88px;"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        test-id="drawerButton"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 logo css-p7bec2-MuiTypography-root"
      >
        Ruth Chernous
      </div>
      <div
        class="MuiBox-root css-130f8nx"
      >
        <a
          href="/"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium link linkActive css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Home
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
           
        </a>
        <a
          href="/about"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium link null css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            About
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
           
        </a>
      </div>
    </div>
  </nav>
  <nav
    class="MuiBox-root css-0"
  />
</div>
`;
